src/components/Movement.ts

```diff
@@ -25,11 +29,56 @@ export function selectNextMove(self: Actor, playBoard: Cell[][]): Cell {
   }
 }
 
-export function isPossibleDirection(actor: Actor, point: number[], playBoard: Cell[][]): boolean {
-  return isGroundCell(actor.position.x + point[0], actor.position.y + point[1], playBoard);
+export const basicDirections = [
+  {
+    action: MovementAction.Left, point: new Point(-1, 0),
+  },
+  {
+    action: MovementAction.Up, point: new Point(0, -1),
+  },
+  {
+    action: MovementAction.Right, point: new Point(1, 0),
+  },
+  {
+    action: MovementAction.Down, point: new Point(0, 1),
+  },
+];
+
+export function getDirectionsForThePoint(point: Point): Point[] {
+  return basicDirections.map((e) => new Point(point.x + e.point.x, point.y + e.point.y));
+}
+
+export function getRandomDirection(availableDirections: MovementAction[]): MovementAction {
+  return availableDirections[Math.floor(Math.random() * availableDirections.length)];
+}
+
+export function basicBfs(pb: PlayBoard, root: Point, depth: number): Point[] {
+  const queue: Point[] = [];
+  const explored: Point[] = [];
+
+  queue.unshift(root);
+
+  while (queue.length > 0) {
+    const current = queue.pop();
```

Comment:
Can be declared as `queue.pop()!` to avoid referring to this as `current!` below.